### PR TITLE
Support multiple files for datagrid fields

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

This PR allows file fields in Data Grid Rows, so that multiple files can be uploaded in a single field:

---

**☝️ NOTE:** The DGF needs to be configured with `allow_insert=False`, e.g.:

```python
class ICertificateRecord(Interface):
    """Record schema for certificates
    """
    directives.widget("certificate", DataGridRowFileWidget)
    certificate = NamedBlobFile(
        title=u"Certificate",
        required=True,
    )

class ITracerSchema(model.Schema):
    """Schema interface
    """

    directives.widget("certificates",
                      DataGridWidget,
                      allow_insert=False,
                      allow_reorder=True)
    certificates = schema.List(
        title=_(
            u"title_tracer_certificates",
            default=u"Certificates"
        ),
        description=_(
            u"description_tracer_certificates",
            default=u"Please upload all certificates for this Tracer"
        ),
        value_type=DataGridRow(schema=ICertificateRecord),
        required=False,
        missing_value=[],
    )
```

💡 Tip:
It is also possible to download these files directly from the field itself, by
`..content/++field++certificates/@@download/0/certificate`
 
## Current behavior before PR

There is no usable way to allow multiple files in a single field in SENAITE

## Desired behavior after PR is merged

Multiple fields can be uploaded in Data Grid Field Rows

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/z3cform/widgets/datagrid/datagridrow_file_input.pt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                             title filename;"/>
+        <a tal:content="filename"
+           tal:attributes="href download_url">
+          Filename
+        </a>
+        <span class="discreet"> &mdash;<tal:doc_type condition="doc_type">
+            <span tal:replace="doc_type"
+                  i18n:translate="">ContentType</span>,</tal:doc_type>
+            <span tal:define="sizekb view/file_size" tal:replace="sizekb"
+                  i18n:translate="">100</span>
```
Comment (xispa) [inline src/senaite/core/z3cform/widgets/datagrid/datagridrow_file_input.pt]: Remove `i18n:translate`

File: src/senaite/core/z3cform/widgets/datagrid/datagridrow_file_input.pt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+             tal:attributes="src icon;
+                             alt doc_type;
+                             title filename;"/>
+        <a tal:content="filename"
+           tal:attributes="href download_url">
+          Filename
+        </a>
+        <span class="discreet"> &mdash;<tal:doc_type condition="doc_type">
+            <span tal:replace="doc_type"
+                  i18n:translate="">ContentType</span>,</tal:doc_type>
```
Comment (xispa) [inline src/senaite/core/z3cform/widgets/datagrid/datagridrow_file_input.pt]: Remove `i18n:translate`

File: src/senaite/core/z3cform/widgets/datagrid/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+```python
+from plone.autoform import directives
+from plone.namedfile.field import NamedBlobFile
+from plone.supermodel import model
+from senaite.core import senaiteMessageFactory as _
+from senaite.core.schema.fields import DataGridField
+from senaite.core.schema.fields import DataGridRow
+from senaite.core.z3cform.widgets.datagrid import DataGridRowFileWidget
+from senaite.core.z3cform.widgets.datagrid import DataGridWidget
```
Comment (xispa) [inline src/senaite/core/z3cform/widgets/datagrid/README.md:30]: Missing imports: 

```python:
from zope import schema
from zope.interface import Interface
```

**Review by xispa — CHANGES_REQUESTED** [review]
Fantastic, it works like a charm!

I've left three minor comments. Besides, in view mode, the widget is rendered like this, with filenames in plain text:

Captura de 2024-07-15 16-13-22

I think would be better if we render links for direct download. What do you think?

**Comment by ramonski** [comment]
Hey Jordi, thanks for your review! I addressed all the points you mentioned and it is ready for review again:)

**Review by xispa — APPROVED** [review]

**Pull request merged**